modules/rest/src/main/java/eu/cloudnetservice/modules/rest/util/QRCodeBuilder.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019-2022 CloudNetService team & contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.cloudnetservice.modules.rest.util;
+
+import com.google.common.base.Joiner;
+import com.google.zxing.BarcodeFormat;
+import com.google.zxing.EncodeHintType;
+import com.google.zxing.MultiFormatWriter;
+import com.google.zxing.WriterException;
+import com.google.zxing.client.j2se.MatrixToImageWriter;
+import com.google.zxing.common.BitMatrix;
+import eu.cloudnetservice.common.document.gson.JsonDocument;
+import eu.cloudnetservice.common.log.LogManager;
+import eu.cloudnetservice.node.Node;
+import eu.cloudnetservice.node.config.ExternalURLInformation;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.NonNull;
+
+public final class QRCodeBuilder {
+
+  private final JsonDocument content = JsonDocument.newDocument();
+  private boolean appSchema = false;
+
+  public static QRCodeBuilder builder() {
```

Comment:
Add static builder method to create a copy of another builder

Reply:
Makes which sense ?